src/WebJobs.Script.WebHost/Diagnostics/LinuxContainerEventGenerator.cs

```diff
@@ -37,6 +38,18 @@ public LinuxContainerEventGenerator(IEnvironment environment, Action<string> wri
 
         public static string AzureMonitorEventRegex { get; } = $"{ScriptConstants.LinuxAzureMonitorEventStreamName} (?<Level>[0-6]),(?<ResourceId>[^,]*),(?<OperationName>[^,]*),(?<Category>[^,]*),(?<RegionName>[^,]*),\"(?<Properties>[^,]*)\",(?<ContainerName>[^,\"]*),(?<TenantId>[^,\"]*),(?<EventTimestamp>[^,]+)";
 
+        public static LinuxContainerEventGenerator GetLinuxContainerEventGenerator
+        {
+            get
+            {
+                if (_eventGenerator == null)
```

Comment:
this might not be thread safe

use Lazy<T> instead or initialize in a static constructor.
[https://csharpindepth.com/Articles/Singleton](https://csharpindepth.com/Articles/Singleton)